README_local_MacOS.md

```diff
@@ -0,0 +1,122 @@
+# Example Local Dev Environment Setup
+
+## MacOS
+
+* `Go` version from
+  [go.mod](https://github.com/gravitational/teleport/blob/master/go.mod#L3)
+  
+  * Follow [official instructions](https://go.dev/doc/install) to install `Go`
+    * **On an M1 Mac, download ARM64 installer from https://go.dev/dl/**
+    * Download the installer for `<version from go.mod>`  
+    * After installing, don't forget to `export PATH="/usr/local/go/bin:$PATH"` in `~/.zprofile`
+    * If you need other go versions, see https://go.dev/doc/manage-install
+      * You will need to add `export PATH="$HOME/go/bin:$PATH"` to the `~/.zprofile`
+
+  * Or install required version of `Go` with homebrew:
+
+  ```shell
+  # if we are not on the latest, you might need to
+  # $ brew install go@<version from go.mod>, i.e. 1.16
+  #
+  # check which version will be installed by running
+  # $ brew info go
+  
+  $ brew install go
+  ````
+
+* `Rust` and `Cargo` version from
+  [build.assets/Makefile](https://github.com/gravitational/teleport/blob/master/build.assets/Makefile#L21)
+  (search for RUST_VERSION):
+
+  * Follow [official instructions](https://www.rust-lang.org/tools/install) to install `rustup`
+  *  Or install with homebrew:
+  
+  ```shell
+  $ brew install rustup
+  ```
+  
+  * Initialize Rustup
+  
+  ```shell
+  $ rustup-init
+  # accept defaults
+  #
+  # Once command finishes successfully, you might need to add
+  # 
+  # export PATH="$HOME/.cargo/bin:$PATH"
+  # 
+  # into ~/.zprofile and run
+  # 
+  # $ . ~/.zprofile
+  # 
+  # or open a new shell
+  ```
+  
+  * If on M1 Mac, add `x86_64` target 
+  
+  ```shell
+  $ rustup target add x86_64-apple-darwin
+  ```
+
+  * Install the required version
+  
+  ```shell
+  $ rustup toolchain install <version from build.assets/Makefile>
+  # 
+  # In addition to the above, if on M1 Mac, also run
+  #
+  # $ rustup toolchain install <version from build.assets/Makefile> --target x86_64-apple-darwin
+  $ cd <teleport.git>
+  $ rustup override set <version from build.assets/Makefile>
+  $ rustc --version                                                                                                                                                                  ─╯
+  rustc <version from build.assets/Makefile> (db9d1b20b 2022-01-20)
+  ```
+
+* To install `libfido2` (pulls `openssl 1.1.1` as dependency)
+ 
+  ```shell
+  $ brew install libfido2
```

Comment:
Suggestion: Add the `export PKG_CONFIG_PATH` here and remove from main README?

Reply:
export PKG_CONFIG_PATH is part of the build instructions. This is more of a how to setup the environment. I don't think it belongs here.